src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -14,23 +16,29 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+@NoArgsConstructor
 public class RouterFileSystem extends FileSystem {
 
-    public static final Logger LOG = LoggerFactory.getLogger(RouterFileSystem.class);
+    private static final Logger LOG = LoggerFactory.getLogger(RouterFileSystem.class);
     private static final String DEFAULT_FS_CONF_PATTEREN = "^routerfs\\.default\\.fs\\.(?<fromScheme>[-a-z0-9_]*)";
     private static final String DEFAULT_FS_SCHEME_REGEX_GROUP_NAME = "fromScheme";
     private static final String DEFAULT_FS_SCHEME_SUFFIX = "-default";
     private static final String DEFAULT_FS_CONF_PREFIX = "routerfs.default.fs";
 
     private PathMapper pathMapper;
+    private Path workingDirectory;
+    private URI uri;
+
+    public RouterFileSystem(PathMapper pathMapper) {
```

Comment:
Do we need this constructor for testing purposes?

Reply:
Unfortunately... If only we could implement dependency injection according to our rules and not Hadoop's...